transaction/transaction.go

```diff
@@ -806,3 +807,117 @@ func byte32FromBase64(in string) (out [32]byte, err error) {
 	copy(out[:], slice)
 	return
 }
+
+var emptySchema = types.StateSchema{}
+
```

Comment:
Could you move these above `AssignGroupID` so that they're next to all the other `Make*` functions?

Also, we seem to have `Make*FlatFee` for all of the other make functions, could you add those here too?

Reply:
Also please follow the same naming convention as the other constructors, remove `Unsigned` from the new methods.

Reply:
I think there are implementations in a `future` package somewhere that uses the "suggested params" approach, maybe we should only implement the new form of these? Please check with @EvanJRichard about that.

Reply:
+1 to also adding a FlatFee option.

Yes - matching py-sdk, the transaction building functions that use the `SuggestedParams` are in package `future` to avoid breaking backwards compatibility. Because this PR's proposed changes don't break backwards compatibility, they _might_ be okay here in `transaction`. But either way I'd really love to see a version in `future` that accepts the `SuggestedParams` object, my understanding is we want to start nudging user devs towards using `SuggestedParams`.  

I like this new way of building transactions (make a transaction + set a header) but it does break the pattern of how the other transaction building functions work, so if we choose to stick with this method I think we should have extra documentation/examples in the `README`